Document example of a tooltip for svg that works on line/area plots

djay opened the pull request:
## PR Summary

## PR Checklist

<!-- Please mark any checkboxes that do not apply to this PR as [N/A]. -->
**Tests and Styling**
- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).

**Documentation**
- [ ] New features are documented, with examples if plot related.
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).

<!--
Thank you so much for your PR!  To help us review your contribution, please
consider the following points:

- A development guide is available at https://matplotlib.org/devdocs/devel/index.html.

- Help with git and github is available at
  https://matplotlib.org/devel/gitwash/development_workflow.html.

- Do not create the PR out of main, but out of a separate branch.

- The PR title should summarize the changes, for example "Raise ValueError on
  non-numeric input to set_xlim".  Avoid non-descriptive titles such as
  "Addresses issue #8576".

- The summary should provide at least 1-2 sentences describing the pull request
  in detail (Why is this change required?  What problem does it solve?) and
  link to any relevant issues.

- If you are contributing fixes to docstrings, please pay attention to
  http://matplotlib.org/devel/documenting_mpl.html#formatting.  In particular,
  note the difference between using single backquotes, double backquotes, and
  asterisks in the markup.

We understand that PRs can sometimes be overwhelming, especially as the
reviews start coming in.  Please let us know if the reviews are unclear or
the recommended next step seems overly demanding, if you would like help in
addressing a reviewer's comments, or if you have been waiting too long to hear
back on your PR.
-->

oscargus commented on examples/user_interfaces/svg_tooltip_line.py:
```diff
@@ -0,0 +1,221 @@
+with timeseries plots showing the values at the x position where the 
+mouse is hovering.
+
+The tooltip is made by dynamically inserting a SVG forignObject with a 
```
```suggestion
The tooltip is made by dynamically inserting a SVG ``foreignObject`` with a 
```

melissawm commented:
👋🏻 Hello @djay - are you interested in moving this forward? Can we help you with this? Cheers!

djay commented:
@melissawm yes. I got stuck. It works great for my usecase but working out how to get the right measurements to get it aligned generally and for this test example proved hard

djay committed:
Add SVG crosshair tooltip example for line, area, and bar plots

When the user hovers an SVG of a matplotlib figure, a vertical line
tracks the cursor, a coloured dot snaps to each series nearest data
point, and a small table shows the values. The legend is hidden in
the active subplot while hovering so the tooltip can take its place.

Supports Line2D (ax.plot), PolyCollection (ax.fill_between,
ax.stackplot), and BarContainer (ax.bar), and works across multiple
subplots -- each series dot only moves while the cursor is in that
series axes.

The data-to-SVG coordinate transform is computed in Python via
ax.transData and embedded into the SVG as JSON, so the JavaScript
side stays small (~70 lines) and does not have to know anything
about axis ranges, log scales, date locators, or axis padding.

Includes a Playwright-based test that loads the SVG in a headless
browser, simulates mouse moves, and verifies the hover dot lands on
the matplotlib-drawn path for each series. Playwright is a soft
dependency -- the tests are skipped if it is not installed.

melissawm commented:
Hi @djay, nice to see you back here! Don't forget to set this as "Ready for review" when you are done. Cheers!

djay commented:
@melissawm done
